src/main/java/model/output/DataTableWriter.java

```diff
@@ -59,6 +65,42 @@ public void write(DataTable dataTable,
 		}
 	}
 
+	public void writeCSV(DataTable table, File file) throws IOException {
+		logger.log(Level.INFO, "writing to CSV format");
+		FileWriter fileWriter = null;
+		CSVPrinter csvFilePrinter = null;
+		//Create the CSVFormat object with "\n" as a record delimiter
+		CSVFormat csvFileFormat = CSVFormat.DEFAULT.withRecordSeparator(NEW_LINE_SEPARATOR);
+
+		try {
+			//initialize FileWriter object
+			fileWriter = new FileWriter(file);
+			//initialize CSVPrinter object
+			csvFilePrinter = new CSVPrinter(fileWriter, csvFileFormat);
+			List<String> rowValues;
+			for (DataRow row : table.getRows()) {
+				rowValues = new ArrayList<>();
+				for (DataColumn col : row.getColumns()) {
+					DataValue value = row.getValue(col);
+					rowValues.add(formatValue(value));
+				}
+				csvFilePrinter.printRecord(rowValues);
+			}
+		} catch (IOException e) {
+			logger.log(Level.SEVERE, "Error writing file", e);
+			throw e;
+		} finally {
+			try {
+				fileWriter.flush();
+				fileWriter.close();
+				csvFilePrinter.close();
+			} catch (IOException e) {
+				logger.log(Level.SEVERE, "Error while flushing/closing fileWriter/csvPrinter !");
+				e.printStackTrace();
+			}
+		}
+		}
```

Comment:
De indentation klopt niet en hij is best wel heel lang. Daarnaast kan je try with resources gebruiken om dingen mooier te maken.